R/read-image.R

```diff
@@ -11,7 +11,7 @@ read_image <- function(path) {
 
 get_file <- function(path) {
   if (is_url(path)) {
-    RCurl::getBinaryURL(path)
+    curl::curl_fetch_memory(path)$content
```

Comment:
Could you explain why `curl_fetch_memory()` instead of just `curl()` or `curl_download()`? I'm not that familiar with the curl package, so I don't know which is the right choice. Also, I think we don't have any test that actually tests this function. We should add one.

Reply:
The test would be a visual test here:
https://github.com/wilkelab/gridtext/blob/master/tests/testthat/test-richtext-grob.R
of simple HTML with an `<img>` tag, probably using this image:
https://github.com/wilkelab/gridtext/blob/0baa935a5d1385a2beb4d32df09b880501c92b22/tests/testthat/test-raster-box.R#L2

Reply:
The old `RCurl::getBinaryURL()` returned the raw binary data from the url. `curl_download()` could also do this but would require a temporary local file. That's why I used `curl_fetch_memory()`. It returns mostly the same as `http::GET()`. I use it in multiple packages and it would to use more than just the raw content, e.g. add a check for http status code in the fetched data for example.

Reply:
I wonder how much sense that makes because this `logo_file` wouldn't be from an url which means we wouldn't check the curl functionality at all. I planned to download [this image](https://upload.wikimedia.org/wikipedia/commons/thumb/2/27/20140427Iris_versicolor1.jpg/320px-20140427Iris_versicolor1.jpg) and save it in tests/figs and compare the output of `read_image` using the url and the local file.
This test should probably be skipped on cran in case we are afraid the url might break at some point

Reply:
Sure, sounds good.

Reply:
I am going to mark this as resolved now because questions are (hopefully) answered and the test is implemented